src/jit/hwintrinsic.cpp

```diff
@@ -3,9 +3,44 @@
 // See the LICENSE file in the project root for more information.
 
 #include "jitpch.h"
+#include "hwintrinsic.h"
 
 #ifdef FEATURE_HW_INTRINSICS
 
+static const HWIntrinsicInfo hwIntrinsicInfoArray[] = {
+// clang-format off
+#if defined(_TARGET_XARCH_)
+#define HARDWARE_INTRINSIC(id, name, isa, ival, size, numarg, t1, t2, t3, t4, t5, t6, t7, t8, t9, t10, category, flag) \
+    {NI_##id, name, InstructionSet_##isa, ival, size, numarg, t1, t2, t3, t4, t5, t6, t7, t8, t9, t10, category, static_cast<HWIntrinsicFlag>(flag)},
+#include "hwintrinsiclistxarch.h"
+#elif defined (_TARGET_ARM64_)
+#define HARDWARE_INTRINSIC(isa, name, ival, size, numarg, t1, t2, t3, t4, t5, t6, t7, t8, t9, t10, category, flag) \
+    {NI_##isa##_##name, #name, InstructionSet_##isa, ival, size, numarg, t1, t2, t3, t4, t5, t6, t7, t8, t9, t10, category, static_cast<HWIntrinsicFlag>(flag)},
```

Comment:
I changed ARM to use this because that is the pattern we were already following for naming things. Might be nice to also update x86 to do the same as it makes the `#defines` in `hwintrinsiclist*.h` smaller, but I felt that was a separate PR.